python/cuspatial/cuspatial/tests/binpreds/test_binpred_test_dispatch.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from functools import wraps
+
+import pandas as pd
+import pytest
+from binpred_test_dispatch import predicate, simple_test  # noqa: F401
+
+"""Decorator function that xfails a test if an exception is throw
+by the test function. Will be removed when all tests are passing."""
+
+
+def xfail_on_exception(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            pytest.xfail(f"Xfailing due to an exception: {e}")
+
+    return wrapper
+
+
+# In the below file, all failing tests are recorded with visualizations.
+out_file = open("test_binpred_test_dispatch.log", "w")
+
+
+@xfail_on_exception  # TODO: Remove when all tests are passing
+def test_simple_features(
+    predicate,  # noqa: F811
+    simple_test,  # noqa: F811
+    predicate_passes,
+    predicate_fails,
+    feature_passes,
+    feature_fails,
+    request,
+):
+    """Parameterized test fixture that runs a binary predicate test
+    for each combination of geometry types and binary predicates.
+
+    Uses four fixtures from `conftest.py` to store the number of times
+    each binary predicate has passed and failed, and the number of times
+    each combination of geometry types has passed and failed. These
+    results are saved to CSV files after each test.
+
+    Uses the @xfail_on_exception decorator to mark a test as xfailed
+    if an exception is thrown. This is a temporary measure to allow
+    the test suite to run to completion while we work on fixing the
+    failing tests.
+
+    Parameters
+    ----------
+    predicate : str
+        The name of the binary predicate to test.
+    simple_test : tuple
+        A tuple containing the name of the test, a docstring that
+        describes the test, and the left and right geometry objects.
+    predicate_passes : dict
+        A dictionary fixture containing the number of times each binary
+        predicate has passed.
+    predicate_fails : dict
+        A dictionary fixture containing the number of times each binary
+        predicate has failed.
+    feature_passes : dict
+        A dictionary fixture containing the number of times each combination
+        of geometry types has passed.
+    feature_fails : dict
+        A dictionary fixture containing the number of times each combination
+        of geometry types has failed.
+    request : pytest.FixtureRequest
+        The pytest request object. Used to print the test name in
+        diagnostic output.
+    """
+    try:
+        (lhs, rhs) = simple_test[2], simple_test[3]
+        gpdlhs = lhs.to_geopandas()
+        gpdrhs = rhs.to_geopandas()
+        pred_fn = getattr(lhs, predicate)
+        got = pred_fn(rhs)
+        gpd_pred_fn = getattr(gpdlhs, predicate)
+        expected = gpd_pred_fn(gpdrhs)
+        assert (got.values_host == expected.values).all()
+
+        # The test is complete, the rest is just logging.
```

Comment:
I appreciate the logging to file part, which can prove to be useful when bulk debugging a new feature of hundreds of failures. But when this system is online in CI, it would also be helpful to have a "CI mode" that throws the result to stdout. What do you think?

Reply:
Because there will be more binary predicate PRs to cover complex (multi-) geometry types, how would you feel about filing an issue for this feature? I'm certainly fine with implementing it but I'd rather not make it a dependency of #1064 that holds it up and needs to be merged.

Reply:
That's totally fine with me